apps/editing-toolkit/editing-toolkit-plugin/common/disable-heic-images.js

```diff
@@ -0,0 +1,38 @@
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { createElement } from '@wordpress/element';
+import { addFilter } from '@wordpress/hooks';
+import { __ } from '@wordpress/i18n';
+
+const DisableHEICImages = createHigherOrderComponent( ( MediaPlaceholder ) => {
+	return ( props ) => {
+		props.onFilesPreUpload = ( files ) => {
+			if ( files ) {
+				const error = __(
+					'HEIC images are not viewable in the editor. Please convert to a JPG, PNG, or GIF and try again.',
+					'full-site-editing'
+				);
+				Object.values( files ).filter( ( file ) => {
+					const filename = file.name;
+					const extension = filename
+						.substring( filename.lastIndexOf( '.' ) + 1, filename.length )
+						.toLowerCase();
+
+					if (
+						'image/heic' === file.type ||
+						'image/heif' === file.type ||
```

Comment:
Optional: it seems checking the type [is redundant](https://developer.mozilla.org/en-US/docs/Web/API/File/type#:~:text=note%3A%20based%20on%20the%20current%20implementation) if the extension is checked.